### PR TITLE
Add job_name override option to bake command

### PR DESCRIPTION
**Pull request by cisaacstern**
This supersedes #17 as a way to pass backend app context information into Dataflow: via the job_name.

This should go in after #14 & #15 (or instead of them), as it incorporates changes from those PRs.

I will circle back to add tests shortly. 🙏

### CODE REVIEW

File: pangeo_forge_runner/commands/bake.py [hunk 1]
```diff
@@ -55,6 +55,28 @@ class Bake(BaseCommand):
...
+    )
+
+    job_name = Unicode(
+        None,
+        allow_none=True,
+        config=True,
+        help="""
+        Optionally pass a custom job name for the job run.
+
+        If empty, a name will be generated for the job below.
```
Comment (yuvipanda) [inline pangeo_forge_runner/commands/bake.py:76]:
```suggestion
        If empty, a unique name will be generated for the job.
```

This documentation is also read by end users, so 'below' doesn't make sense outside of the code.

File: pangeo_forge_runner/commands/bake.py [hunk 2]
```diff
@@ -97,19 +123,18 @@ def start(self):
 
             for name, recipe in recipes.items():
                 # Unique name for running this particular recipe.
-                # FIXME: Should include the name of repo / ref as well somehow
-                job_name = (
-                    f"{name}-{recipe.sha256().hex()}-{int(datetime.now().timestamp())}"
-                )
+                if not self.job_name:
+                    # FIXME: Should include the name of repo / ref as well somehow
+                    self.job_name = f"{name}-{recipe.sha256().hex()}-{int(datetime.now().timestamp())}"
```
Comment (yuvipanda) [inline pangeo_forge_runner/commands/bake.py:128]:
```suggestion
                    job_name = f"{name}-{recipe.sha256().hex()}-{int(datetime.now().timestamp())}"
                else:
                    job_name = self.job_name
```

Easier to see here that `job_name` is always not None even if `self.job_name` is.

File: pangeo_forge_runner/commands/bake.py [hunk 3]
```diff
@@ -97,19 +123,18 @@ def start(self):
...
 
                 recipe.storage_config = StorageConfig(
-                    target_storage.get_forge_target(job_name=job_name),
-                    input_cache_storage.get_forge_target(job_name=job_name),
-                    metadata_cache_storage.get_forge_target(job_name=job_name),
+                    target_storage.get_forge_target(job_name=self.job_name),
```
Comment (yuvipanda) [inline pangeo_forge_runner/commands/bake.py:131]:
```suggestion
                    target_storage.get_forge_target(job_name=job_name),
```

File: pangeo_forge_runner/commands/bake.py [hunk 4]
```diff
@@ -97,19 +123,18 @@ def start(self):
...
+                    input_cache_storage.get_forge_target(job_name=self.job_name),
```
Comment (yuvipanda) [inline pangeo_forge_runner/commands/bake.py:132]:
```suggestion
                    input_cache_storage.get_forge_target(job_name=job_name),
```

File: pangeo_forge_runner/commands/bake.py [hunk 5]
```diff
@@ -97,19 +123,18 @@ def start(self):
...
+                    metadata_cache_storage.get_forge_target(job_name=self.job_name),
```
Comment (yuvipanda) [inline pangeo_forge_runner/commands/bake.py:133]:
```suggestion
                    metadata_cache_storage.get_forge_target(job_name=job_name),
```

**Review by yuvipanda — COMMENTED** [review]
Left some style comments but LGTM! Happy to merge once we add a simple test.

**Comment by yuvipanda** [comment]
@cisaacstern can you add a test?

**Comment by cisaacstern** [comment]
Yep working on that now!

**Comment by cisaacstern** [comment]
@yuvipanda, just added a test.

File: pangeo_forge_runner/commands/bake.py [hunk 6]
```diff
@@ -66,6 +66,17 @@ class Bake(BaseCommand):
...
     )
 
+    job_name = Unicode(
+        None,
+        allow_none=True,
+        config=True,
+        help="""
+        Optionally pass a custom job name for the job run.
+
+        If empty, a unique name will be generated for the job.
```
Comment (yuvipanda) [inline pangeo_forge_runner/commands/bake.py:76]:
```suggestion
        If `None` (the default), a unique name will be generated for the job.
```

**Comment by yuvipanda** [comment]
Thanks, @cisaacstern!

**Comment by cisaacstern** [comment]
Incredible, @yuvipanda! When you get a moment could you release 0.5?

**Pull request merged**